### PR TITLE
Rough draft of Gitea to Gitea migration.

### PR DESCRIPTION
**Pull request by j-be**
Add a GiteaDownloader implemenatation for Gitea, allowing Gitea -> Gitea Migrations.

[ ] Find out how to get or default missing fields
[ ] Work on unit tests

refs #8380

### CODE REVIEW

**Comment by 6543** [comment]
IsLocked property I would set to false if I cant get the inf - however for v1.13 we will have this info soon :)
https://github.com/go-gitea/gitea/pull/11708

**Comment by j-be** [comment]
Ok, seems like I'm making progress. But I found another property I'm missing: `base.PulRequest.OriginalNumber`. I see it is used in `gitlab.go`, but not in `github.go`. @6543 shall I just hardcode it to 0?

**Comment by 6543** [comment]
@j-be just ignore OriginalNumber - as github do

File: modules/migrations/gitea_downloader.go [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+		reactions, _ := g.GetReactionsToIssue(issueNumber)
+		var assignee string
+		if pr.Assignee != nil {
+			assignee = pr.Assignee.UserName
+		}
+
+		allPRs = append(allPRs, &base.PullRequest{
+			Number:         pr.Index,
+			OriginalNumber: 0,
```
Comment (6543) [inline modules/migrations/gitea_downloader.go]:
```suggestion
```

**Comment by j-be** [comment]
Done. What do you think about line 503 `currentReviewComments = make([]*base.ReviewComment, 0)`? Does that make any sense? I'm fairly new to Go and usually do Java, where you better use empty lists instead of `null`. Does that apply to Go, too?

**Comment by 6543** [comment]
If you itterate throu a null list nothing will hapen ...

**Comment by lunny** [comment]
@j-be that's not efficent and we need a refactor but I think that's not the business of this PR.

**Comment by 6543** [comment]
what's the matter?

**Comment by 6543** [comment]
@j-be sorry that I have created a similar pr ->  #12657

hope you dont mid :)
I would be happy if you could test 12657 and report issues have code suggestions :)

still a bug thanks to the work you have done so far - it was quit the same result as I got ;)

**Pull request closed without merging**